tests/test_mongodb_fname_restrictions.py

```diff
@@ -0,0 +1,285 @@
+            changed_ids.add(object_id)
+            client["simple_db"]["simple_coll_2"].delete_one({'_id': object_id})
+
+            # Update two documents for each collection
+
+            # curor objects do not support negative indicies so set indicies for last two records
+            num_records = client['simple_db']['simple_coll_1'].find().count()
+            last_index = num_records - 1
+            sec_last_index = num_records -2
+
+            object_id = client['simple_db']['simple_coll_1'].find()[sec_last_index]['_id'] # int.field 48
+            changed_ids.add(object_id)
+            client["simple_db"]["simple_coll_1"].update_one({'_id': object_id},{'$set': {'int.field': -1}})
+
+            object_id = client['simple_db']['simple_coll_1'].find()[last_index]['_id'] # int.field 49
+            changed_ids.add(object_id)
+            client["simple_db"]["simple_coll_1"].update_one({'_id': object_id},{'$set': {'int.field': -1}})
+
+            num_records = client['simple_db']['simple_coll_2'].find().count()
+            last_index = num_records - 1
+            sec_last_index = num_records - 2
+
+            object_id = client['simple_db']['simple_coll_2'].find()[sec_last_index]['_id'] # int.field 98
+            changed_ids.add(object_id)
+            client["simple_db"]["simple_coll_2"].update_one({'_id': object_id},{'$set': {'int.field': -1}})
+
+            object_id = client['simple_db']['simple_coll_2'].find()[last_index]['_id'] # int.field 99
+            changed_ids.add(object_id)
+            client["simple_db"]["simple_coll_2"].update_one({'_id': object_id},{'$set': {'int.field': -1}})
+
+            # Insert two documents for each collection
+            last_index = client['simple_db']['simple_coll_1'].find().count()
+            client["simple_db"]["simple_coll_1"].insert_one({"int.field": 50, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_1"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
+            last_index += 1 # inserting a new item
+            client["simple_db"]["simple_coll_1"].insert_one({"int.field": 51, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_1"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
+            last_index = client['simple_db']['simple_coll_2'].find().count()
+            client["simple_db"]["simple_coll_2"].insert_one({"int.field": 100, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_2"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
+            last_index += 1
+            client["simple_db"]["simple_coll_2"].insert_one({"int.field": 101, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_2"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
```

Comment:
Why the insert for the `$` fields, and the update for the `.` fields? Is this because of the client limitations we discussed?

Reply:
Updated to insert and update for both special characters (although the '.' is not supported for updates in this mongo version(documented in TDL20088).